R/test-that.R

```diff
@@ -167,6 +167,11 @@ test_code <- function(code, env, reporter = NULL, skip_on_empty = TRUE) {
           }
         },
         expectation = handle_expectation,
+        packageNotFoundError = function(e) {
```

Comment:
this looks like a great & reasonable compromise -- minimal implementation & fits well with existing approach that always sets `!on_cran` in package GHA.

one possible concern is that it doesn't make any effort to distinguish "package missing in current environment" from "you forgot to add a package to your `Suggests`". I guess `R CMD check` is inspecting `tests/` files for `pkg::` usage so it should be caught anyway.